WooCommerce/WooCommerceTests/ViewRelated/Orders/Order Creation/ProductBundles/ConfigurableBundleProductViewModelTests.swift

```diff
@@ -341,7 +341,8 @@ final class ConfigurableBundleProductViewModelTests: XCTestCase {
                                                            onConfigure: { _ in })
 
         // The products are loaded async before the bundle item view models are set.
-        waitUntil {
+        // This test occasionally fails on CI due to slowness, cannot reproduce locally.
+        waitUntil(timeout: 10.0) {
```

Comment:
😨 it passes locally with a timeout of 1 or less, and I cannot see any reason right away of why would fail in CI. 

Just wondering here if the session we're injecting in the mock would have any effect into this. We're using `.makeForTesting()` while in other tests we use`.testingInstance` . I'd say no, but also we don't need the whole session details to init this class.

Outside of this PR and as a side note, I don't think `makeForTesting` name is very good, there's no clear difference with `.testingInstance`

Reply:
I was banging my head with this test. I haven't tried playing with `testingInstance` vs `makeForTesting`, I probably should. 

This works fast on local machine and incredibly slow on the CI...

Reply:
I tried to make this change just for experiment purpose and will check tomorrow.

Reply:
It still runs in over 5 seconds... It's a true mystery to me. 

<img width="900" alt="image" src="https://github.com/user-attachments/assets/e5ed4839-0d3c-4e76-b78d-ab4b405237b6" />

Reply:
The performance improvements are still solid. I'm gonna merge.